adapters/repos/db/vector/hnsw/compress.go

```diff
@@ -0,0 +1,55 @@
+package hnsw
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	ssdhelpers "github.com/semi-technologies/weaviate/adapters/repos/db/vector/ssdHelpers"
+)
+
+const (
+	centroids = 256
+)
+
+func (h *hnsw) Compress(segments int) error {
+	vec, _ := h.vectorForID(context.Background(), h.nodes[0].id)
```

Comment:
@abdelr I like how we can just infer the dimensions from existing data in the class but if you create a class with `compress: true` and no data, `nodes` will be empty and this will crash.

Reply:
You are right. Actually, should we respond with an error if there is not enough data when compress command comes? I can imagine that it makes no sense to compress without data but even with five vectors it makes no sense either. But then, what is enough data?

Reply:
For now I will check that it exists at least one vector but this still is not enough. It will prevent the crash though.